joomla/joomla-cms
# [4.0] Normalize finder tables and add default values

This is a new version of #13517 which originally was supposed to fix #13509. Since all of my finder work is aimed at 4.0, this PR is, too. I also extended this further and since MSSQL is not supported in 4.0 anymore, I've not adopted those changes.
Since this PR is now open for nearly a year, I also reworked this PR to incorporate all DB changes to the finder tables in one big update file instead of several smaller ones.

## Changes
* Changed all field types to be identical across DB systems and across tables
* Added default values for all relevant fields
* Added indexes where appropriate
* Updated component SQL files
* Merged all DB changes for 4.0 for Finder into one update file

Considering all the changes that happened to finder, a complete re-indexing is necessary. This requirement is not related to this PR, but comes from the removal of the sharding, the new taxonomies and several other changes. This also means that the filters will have to be manually recreated. All these updates can be done on filled tables, but that would make little sense, since we have to run a complete indexing run anyway, thus this PR first clears the tables and then does the updates. That should help with timeout issues for large indexes.

## How to test
I'm quoting the original dev of this PR:
1. SQL code review
2. Test update: Apply patch, manually run the sql query for your Db system, confirm db table, confirm all fine
3. Test install: apply patch, delete configuration.php do a normal install, confirm db table, confirm all fine

@alikon Can you check the Postgres code? Is there a difference between VARCHAR and CHAR in Postgres? Maybe we want to change the language columns from VARCHAR to CHAR? I'm confused because the original author did not change those.
